server/resp/src/main/java/org/infinispan/server/resp/commands/set/SDIFF.java

```diff
@@ -0,0 +1,75 @@
+package org.infinispan.server.resp.commands.set;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.multimap.impl.EmbeddedSetCache;
+import org.infinispan.multimap.impl.SetBucket;
+import org.infinispan.server.resp.Consumers;
+import org.infinispan.server.resp.Resp3Handler;
+import org.infinispan.server.resp.RespCommand;
+import org.infinispan.server.resp.RespRequestHandler;
+import org.infinispan.server.resp.commands.Resp3Command;
+
+import io.netty.channel.ChannelHandlerContext;
+
+/**
+ * {@link} https://redis.io/commands/sdiff/
+ * Returns the members of the set resulting from the difference between
+ * the first set and all the successive sets.
+ * @since 15.0
+ */
+public class SDIFF extends RespCommand implements Resp3Command {
+
+   public SDIFF() {
+      super(-2, 1, -1, 1);
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
+         ChannelHandlerContext ctx,
+         List<byte[]> arguments) {
+      EmbeddedSetCache<byte[], byte[]> esc = handler.getEmbeddedSetCache();
+      // If the diff is with the set itself the result will be an empty set
+      // but we need to go through the process to check all the keys are set type
+      boolean diffItself = arguments.stream().skip(1)
+            .anyMatch(item -> Objects.deepEquals(arguments.get(0), item));
+      // Wrapping to exclude duplicate keys
+      var uniqueKeys = SINTER.getUniqueKeys(handler, arguments);
+      var allEntries = esc.getAll(uniqueKeys);
+      return handler.stageToReturn(
```

Comment:
@rigazilla @jabolina the way I implemented the diff and union differs quite a little between sets and sorted sets.
I don't get all the entries all the time, if there is nothing else to diff (the first is empty) I avoid getting all the keys. But I don't know if there is a real difference for this use case. sorted sets are more complex anyway (with weights and so)

Reply:
I considered to optimize the code like that, but redis returns WRONGTYPE error if a key in the list is not a set even if the diff result is already an empty set; the only way to do that is fetch all the elements.